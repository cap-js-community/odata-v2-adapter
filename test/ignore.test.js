"use strict";

const cds = require("@sap/cds");
const supertest = require("supertest");

const util = require("./_env/util/request");

cds.test(__dirname + "/_env");

let request;

describe("rest", () => {
  beforeAll(async () => {
    await global._init;
    request = supertest(cds.app.server);
  });

  it("GET $metadata", async () => {
    const response = await util.callRead(request, "/odata/v2/ignore/$metadata", {
      accept: "application/xml",
    });
    expect(response.statusCode).toEqual(400);
    expect(response.body).toBeDefined();
    expect(response.text).toMatchSnapshot();
  });

  it("GET data", async () => {
    const response = await util.callRead(request, "/odata/v2/ignore/Header");
    expect(response.statusCode).toEqual(400);
    expect(response.body).toBeDefined();
    expect(response.text).toMatchSnapshot();
  });
});
