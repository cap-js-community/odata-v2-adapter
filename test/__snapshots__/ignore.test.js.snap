// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rest GET $metadata 1`] = `"Service is not exposed as OData V2 protocol"`;

exports[`rest GET data 1`] = `"Service is not exposed as OData V2 protocol"`;
